packages/nvidia_nat_security/tests/middleware/guardrails/test_nemo_guardrails_middleware.py

```
@@ -0,0 +1,316 @@
+    call_next.assert_awaited_once()
+
+
+def test_finalize_guardrails_rejects_missing_policy_source() -> None:
+    """Reject configuration when neither inline policy nor policy root is set."""
+    with pytest.raises(ValueError, match="exactly one of guardrails or guardrails_root"):
+        GuardrailsMiddlewareConfig(functions=["test_fn"])
+
+
+def test_finalize_guardrails_rejects_colang_2() -> None:
+    """Reject Colang 2.x policy at config load."""
+    with pytest.raises(ValueError, match="Colang 2.0 is not supported"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)